libtransact/src/context/error.rs

```diff
@@ -71,3 +73,71 @@ impl From<StateReadError> for ContextManagerError {
         ContextManagerError::StateReadError(err)
     }
 }
+
+#[derive(Debug)]
+pub enum ContextManagerCoreError {
```

Comment:
I wonder if it would make more sense for this to be in the `thread` sub-module, since that's the module that "owns" this error type?